File: backend-php/app/Models/Vehicule.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

/**
 * @OA\Schema(
 *     schema="Vehicule",
 *     required={"immatriculation_part1", "immatriculation_part2", "immatriculation_part3", "marque", "modele", "client_id"},
 *     @OA\Property(property="id", type="integer", readOnly=true),
 *     @OA\Property(property="immatriculation_part1", type="string"),
 *     @OA\Property(property="immatriculation_part2", type="string"),
 *     @OA\Property(property="immatriculation_part3", type="string"),
 *     @OA\Property(property="marque", type="string"),
 *     @OA\Property(property="modele", type="string"),
 *     @OA\Property(property="type_vehicule", type="string"),
 *     @OA\Property(property="carburant", type="string"),
 *     @OA\Property(property="puissance_fiscale", type="string"),
 *     @OA\Property(property="annee_mise_circulation", type="string"),
 *     @OA\Property(property="numero_chassis", type="string"),
 *     @OA\Property(property="kilometrage_compteur", type="integer"),
 *     @OA\Property(property="client_id", type="integer")
 * )
 */
class Vehicule extends Model
{
    protected $table = 'vehicule';
    public $timestamps = false;

    protected $fillable = [
        'immatriculation_part1', 'immatriculation_part2', 'immatriculation_part3',
        'marque', 'modele', 'type_vehicule', 'carburant', 'puissance_fiscale',
        'annee_mise_circulation', 'numero_chassis', 'kilometrage_compteur', 'client_id'
    ];

    public function client()
    {
        return $this->belongsTo(Client::class);
    }

    public function fiches()
    {
        return $this->hasMany(FicheTechnique::class, 'vehicule_id');
    }
}

File: backend/src/main/java/com/app/visite_technique/entity/Vehicule.java
package com.app.visite_technique.entity;

import jakarta.persistence.*;
import java.util.List;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
public class Vehicule {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Immatriculation decomposed based on requirements
    private String immatriculationPart1; // XXXXX
    private String immatriculationPart2; // XX
    private String immatriculationPart3; // FFFFF

    private String marque;
    private String modele;

    // New fields
    private String typeVehicule; // Tourisme, Utilitaire, Taxi, Camion
    private Integer anneeMiseCirculation;
    private String numeroChassis;
    private String carburant; // Essence, Diesel, Hybride, Electrique
    private Integer kilometrageCompteur; // Current mileage if tracked on vehicle, distinct from visit mileage
    private Integer puissanceFiscale;
    private String couleur;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "client_id")
    @JsonIgnoreProperties({ "vehicules", "hibernateLazyInitializer", "handler" })
    private Client client;

    @OneToMany(mappedBy = "vehicule", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnoreProperties({ "vehicule", "hibernateLazyInitializer", "handler" })
    private List<FicheTechnique> fiches;

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getImmatriculationPart1() {
        return immatriculationPart1;
    }

    public void setImmatriculationPart1(String immatriculationPart1) {
        this.immatriculationPart1 = immatriculationPart1;
    }

    public String getImmatriculationPart2() {
        return immatriculationPart2;
    }

    public void setImmatriculationPart2(String immatriculationPart2) {
        this.immatriculationPart2 = immatriculationPart2;
    }

    public String getImmatriculationPart3() {
        return immatriculationPart3;
    }

    public void setImmatriculationPart3(String immatriculationPart3) {
        this.immatriculationPart3 = immatriculationPart3;
    }

    public String getFullImmatriculation() {
        return immatriculationPart1 + "-" + immatriculationPart2 + "-" + immatriculationPart3;
    }

    public String getMarque() {
        return marque;
    }

    public void setMarque(String marque) {
        this.marque = marque;
    }

    public String getModele() {
        return modele;
    }

    public void setModele(String modele) {
        this.modele = modele;
    }

    public Client getClient() {
        return client;
    }

    public void setClient(Client client) {
        this.client = client;
    }

    public String getTypeVehicule() {
        return typeVehicule;
    }

    public void setTypeVehicule(String typeVehicule) {
        this.typeVehicule = typeVehicule;
    }

    public Integer getAnneeMiseCirculation() {
        return anneeMiseCirculation;
    }

    public void setAnneeMiseCirculation(Integer anneeMiseCirculation) {
        this.anneeMiseCirculation = anneeMiseCirculation;
    }

    public String getNumeroChassis() {
        return numeroChassis;
    }

    public void setNumeroChassis(String numeroChassis) {
        this.numeroChassis = numeroChassis;
    }

    public String getCarburant() {
        return carburant;
    }

    public void setCarburant(String carburant) {
        this.carburant = carburant;
    }

    public Integer getKilometrageCompteur() {
        return kilometrageCompteur;
    }

    public void setKilometrageCompteur(Integer kilometrageCompteur) {
        this.kilometrageCompteur = kilometrageCompteur;
    }

    public Integer getPuissanceFiscale() {
        return puissanceFiscale;
    }

    public void setPuissanceFiscale(Integer puissanceFiscale) {
        this.puissanceFiscale = puissanceFiscale;
    }

    public String getCouleur() {
        return couleur;
    }

    public void setCouleur(String couleur) {
        this.couleur = couleur;
    }

    public List<FicheTechnique> getFiches() {
        return fiches;
    }

    public void setFiches(List<FicheTechnique> fiches) {
        this.fiches = fiches;
    }
}
